contracts/Registry.vy

```diff
@@ -605,14 +681,67 @@ def _add_pool(
     self.pool_data[_pool].has_initial_A = _has_initial_A
     self.pool_data[_pool].is_v1 = _is_v1
     self.pool_data[_pool].n_coins = _n_coins
+    self.pool_data[_pool].name = _name
 
     # update public mappings
     self.get_pool_from_lp_token[_lp_token] = _pool
     self.get_lp_token[_pool] = _lp_token
+    self.last_updated = block.timestamp
 
     log PoolAdded(_pool, slice(_rate_method_id, 28, 4))
 
 
+@internal
+def _register_coin(_coin: address):
+    if self.coins[_coin].register_count == 0:
+        self.coins[_coin].index = self.coin_count
+        self.get_coin[self.coin_count] = _coin
+        self.coin_count += 1
+    self.coins[_coin].register_count += 1
+
+
+@internal
+def _register_coin_pair(_coina: address, _coinb: address):
+    if self.coin_swap_register_count[_coina][_coinb] == 0:
```

Comment:
You are reading `self.coin_swap_register_count[_coina][_coinb]` from memory on lines 705 and 709 - can instead assign it to a memory variable to avoid the duplicate lookup.

Same story with `self.coins[_coina].swap_count` on lines 706-708

Reply:
And in other functions.  In general, be aware that vyper doesn't implicitly cache storage variables.  A storage lookup costs us 800 gas and rising, whereas memory access is 3.

Reply:
Ouchie, 800 gas ... I've updated the 4 internal functions I added, they should be legit now, compared to before.